rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,192 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+   * If not decodable, go to 1 and send values based on the previous page.
+   * If decodable, send a request for missing transactions
+   * If Bob's highest Lamport Clock value is higher than the LC value sent in 1, then request transactions by Lamport Clock value:
+     * Lamport Clock start value.
+     * Lamport Clock end value.
+
+4. When receiving a request for transactions, Bob responds with a message including the requested transactions (§4.2.4 and §4.2.5).
+
+5. After adding Bob's transactions to the DAG \(making sure its cryptographic signature is valid\), query the payload if it's missing.
+
+A node MUST make sure to only add transactions of which all previous transactions are present.
+
+The sections below specify the details of the protocol operation and maps it to the gRPC messages. See the section "Protobuf Definition" for a full specification of the protobuf/gRPC contract.
+
+#### 4.2.1 Broadcasting the state
+
+todo:
+The local node's DAG state MUST be broadcast at an interval using the `State` message, by default every 2 seconds. The interval MAY be adjusted by the node operator but MUST conform to the limits \(min/max interval\) defined by the network. It is advised to keep it relatively short because it directly influences the speed by which new transactions are retrieved.
+
+The `State` message contains a `XOR` value, an `LC` value and a `token`. The `token` is scoped to a connection and is required in the response from the peer.
+The `LC` value MUST equals the highest Lamport Clock value of all transaction references included in the `XOR` calculation.
+If no transactions are present, an all-zero `XOR` and `LC` of 0 is sent.
+A node is free to choose how the `token` is created as long as it's able to satisfy the requirements of §4.2.3.
+It MUST be valid for at least 10 seconds.
+
+#### 4.2.2 IBLT response
+
+When a peer receives an `State` message and the `XOR` differs, it SHOULD respond with a `TransactionSet` message.
+The sent `LC` value falls within the bounds of a page.
+The response IBLT MUST be calculated over the transactions leading up to and including that page.
+Next to the IBLT data, the `TransactionSet` message MUST contain the original `LC` value as `LC_req`, a new `LC` value indicating the highest LC value from the peer and the `token` value from the `State` message.
+
+#### 4.2.3 Transaction List Query
+
+For every `TransactionSet` message a node receives, it MUST check the `token` value in the message and compare it to its own administration.
+If the `token` doesn't exist or if the `LC_req` value doesn't match, the node MUST ignore the message.
+
+The IBLT in the `TransactionSet` message contains every transaction of the peer in the LC range of `0-LC_req`. The node MUST lookup/compute the IBLT for this range and subtract it from the IBLT of the peer. Decoding the resulting IBLT will list the transaction refs the peer has and the local node misses. 
```

Comment:
The IBLT received covers the range `0-min(LC, LC_req)`
if `LC < LC_req` we only need to compare to our own IBLT with range `0-LC`. If the LCs are far apart it will not be possible to decode the subtracted IBLTs.